src/libsync/creds/keychainchunk.cpp

```diff
@@ -317,8 +320,122 @@ void ReadJob::slotReadJobDone(QKeychain::Job *incomingJob)
         readJob->deleteLater();
     }
 
-    _isJobRunning = false;
     emit finished(this);
+
+    if (_autoDelete)
+        deleteLater();
+}
+
+/*
+* DeleteJob
+*/
+DeleteJob::DeleteJob(Account *account, const QString &key, const bool &keychainMigration, QObject *parent)
+    : Job(parent)
+{
+    _account = account;
+    _key = key;
+
+    _keychainMigration = keychainMigration;
+}
+
+DeleteJob::DeleteJob(const QString &key, QObject *parent)
+    : DeleteJob(nullptr, key, false, parent)
+{
+#ifdef Q_OS_WIN
+    jobKeyPrependAppName(_key);
+#endif
+}
+
+void DeleteJob::start()
+{
+    _chunkCount = 0;
+    _error = QKeychain::NoError;
+
+    const QString kck = _account ? AbstractCredentials::keychainKey(
+            _account->url().toString(),
+            _key,
+            _keychainMigration ? QString() : _account->id()
+        ) : _key;
+
+    auto *job = new QKeychain::DeletePasswordJob(_serviceName, this);
+#if defined(KEYCHAINCHUNK_ENABLE_INSECURE_FALLBACK)
+    addSettingsToJob(_account, job);
+#endif
+    job->setInsecureFallback(_insecureFallback);
+    job->setKey(kck);
+    connect(job, &QKeychain::Job::finished, this, &KeychainChunk::DeleteJob::slotDeleteJobDone);
+    job->start();
+}
+
+bool DeleteJob::exec()
+{
+    start();
+
+    QEventLoop waitLoop;
+    connect(this, &DeleteJob::finished, &waitLoop, &QEventLoop::quit);
+    waitLoop.exec();
+
+    if (error() == NoError) {
+        return true;
+    }
+
+    _chunkCount = 0;
+    if (error() != EntryNotFound) {
+        qCWarning(lcKeychainChunk) << "DeletePasswordJob failed with" << errorString();
+    }
+    return false;
+}
+
+void DeleteJob::slotDeleteJobDone(QKeychain::Job *incomingJob)
+{
+    // Errors or next chunk?
+    auto *deleteJob = static_cast<QKeychain::DeletePasswordJob *>(incomingJob);
+
+    if (deleteJob) {
```

Comment:
So either you meant qobject_cast above, or you could have checked incomingJob even before the cast (and then an assert would probably be better since receiving a null incomingJob on that slot would mean something went seriously haywire somewhere else).

Reply:
This was part of the original implementation:
https://github.com/nextcloud/desktop/blob/09b0055dc3fe8ffae64dae7af9f522ed12ebdea8/src/gui/creds/webflowcredentials.cpp#L275

Reply:
Still feels wrong, your call if you want to clean it up during that move or if it'll happen later.

Reply:
Oh, that's not the point - just wanted state that I meant nothing with it ;-P

I've already changed it ;-)